.jenkinsci/builders/x64-linux-build-steps.groovy

```diff
@@ -0,0 +1,180 @@
+    + " --network=${env.IROHA_NETWORK}"
+    + " -v /var/jenkins/ccache:${env.CCACHE_DEBUG_DIR}") {
+      utils.ccacheSetup(5)
+      for (compiler in compilerVersions.split(',')) {
+        stage ("build ${compiler}"){
+          build.cmakeConfigure(buildDir, "-DCMAKE_CXX_COMPILER=${compilers[compiler]['cxx_compiler']} \
+            -DCMAKE_C_COMPILER=${compilers[compiler]['cc_compiler']} \
+            -DCMAKE_BUILD_TYPE=${build_type} \
+            -DCOVERAGE=${cmakeBooleanOption[coverage]} \
+            -DTESTING=${cmakeBooleanOption[testing]} \
+            -DFUZZING=${cmakeBooleanOption[fuzzing]} \
+            -DPACKAGE_DEB=${cmakeBooleanOption[packagebuild]} \
+            -DPACKAGE_TGZ=${cmakeBooleanOption[packagebuild]} ${cmakeOptions}")
+          build.cmakeBuild(buildDir, cmakeBuildOptions, parallelism)
+          if (packagebuild) {
+            // if we use several compiler only last build  will saved as iroha.deb and iroha.tar.gz
+            sh "mv ./build/iroha-*.deb ./build/iroha.deb"
+            sh "mv ./build/iroha-*.tar.gz ./build/iroha.tar.gz"
+          }
+        }
+        if (testing) {
+          stage("Test ${compiler}") {
+            coverage ? build.initialCoverage(buildDir) : echo('Skipping initial coverage...')
+            testSteps(buildDir, environment, testList)
+            coverage ? build.postCoverage(buildDir, '/tmp/lcov_cobertura.py') : echo('Skipping post coverage...')
+          }
+        }
+      }
+      stage("Analysis") {
+            cppcheck ? build.cppCheck(buildDir, parallelism) : echo('Skipping Cppcheck...')
+            sonar ? build.sonarScanner(scmVars, environment) : echo('Skipping Sonar Scanner...')
+      }
+      stage('Build docs'){
+        docs ? doxygen.doDoxygen() : echo("Skipping Doxygen...")
+      }
+      stage ('DockerManifestPush'){
+        if (pushDockerTag) {
+          utils.dockerPush(iC, "${platform}-develop-build")
+          DockerManifestPush(iC, "develop-build", environment)
+          if (build_type == 'Release' && scmVars.GIT_LOCAL_BRANCH ==~ /(master|develop|dev)/){
+            sh "cp ./build/iroha.deb docker/release/iroha.deb"
+            iCRelease = docker.build("${env.DOCKER_REGISTRY_BASENAME}:${scmVars.GIT_COMMIT}-${env.BUILD_NUMBER}-release", "--no-cache -f docker/release/Dockerfile ${WORKSPACE}/docker/release")
+            utils.dockerPush(iCRelease, "${platform}-develop")
+            DockerManifestPush(iCRelease, "develop", environment)
+            sh "docker rmi ${iCRelease.id}"
+          }
+        }
+      }
+    }
+  }
+}
+
+def successPostSteps(scmVars, String build_type, List environment) {
+  stage('successPostSteps') {
+    withEnv(environment) {
+      artifacts = load ".jenkinsci/artifacts.groovy"
+      filesToUpload = []
+      platform = sh(script: 'uname -m', returnStdout: true).trim()
+      if (build_type == 'Release' && scmVars.GIT_LOCAL_BRANCH ==~ /(master|develop|dev)/) {
```

Comment:
I would have moved all the logic related to checking branch names, release types and other cases when we need to decide on whether do something or not to Jenkinsfile. Previously there were separate builds for each case (e.g. `x64ReleaseLinuxBuild`) to separate general functions from different build cases described here: https://docs.google.com/spreadsheets/d/1fCt1EPakDI-aBWE17UsZe7_GjTqhaDI6iskuStYb6ok/